language-server/modules/langserver-core/src/test/resources/codeaction/create-variable/config/variableAssignmentRequiredCodeAction38.json

```diff
@@ -16,7 +16,7 @@
                             "character": 4
                         }
                     },
-                    "newText": "object {public isolated function __iterator() returns object {public isolated function next() returns record {| int value; |}? ;} ;} objectResult = "
+                    "newText": "object {int iStart;int iEnd;int iCurrent;public isolated function iterator() returns object {public isolated function next() returns record {| int value; |}? ;} ;} objectResult = "
```

Comment:
Are these fields expected for iterator object ? `int iStart;int iEnd;int iCurrent;`

Reply:
We don't actually need to show fields. Spec does not define an exact implementation for iterator object that we must return for int range expression.
>The result of a range-expr is a new object belonging to the object type Iterable<int,()>

I am returning an object from `createIntRange` method which includes internal `__IntRange` class. If the class has private fields I cannot make the object including the class. I cannot return the `__IntRange` class itself since it is internal.

Reply:
Looks like this code does not compile. Let me check this again?

Reply:
I changed fields qualifiers to public fields because LHS cannot have private fields for object assignability.

Reply:
Even exposing these as public fields looks not so nice imo. I feel we need to omit private fields when generating signature for [ObjectTypeSymbols](https://github.com/ballerina-platform/ballerina-lang/blob/master/compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaObjectTypeSymbol.java#L162). WDYT @pubudu91 @KRVPerera ?

Reply:
I agree that we should omit private field descriptors and method descriptors. This solves the problem of having to expose the fields to support code action.
@pubudu91 Please confirm if this is okay.

Reply:
I prevented non-public fields from contributing to the signature.
b4a1efd330fe21c60de8cab91eaee70f95b37144

Reply:
I will be adding https://github.com/ballerina-platform/ballerina-spec/commit/723b2e0ac26233d77f37e722f2cd74e4205cd92b changes to this PR as well.